### PR TITLE
New: Add service worker

### PR DESCRIPTION
**Pull request by qzhou1607**
## Pull request checklist

Make sure you:

- [X] Signed the [Contributor License Agreement](https://cla.js.foundation/sonarwhal/sonar)
- [X] Followed the [commit message guidelines](https://sonarwhal.com/docs/contributor-guide/contributing/pull-requests.html#commitmessages)

## Short description of the change(s)

Added service worker according to #232

Push notification in progress...

### CODE REVIEW

File: src/hexo/themes/sonarwhal/layout/index.hbs [hunk 1]
```diff
@@ -17,5 +17,6 @@
     {{>footer navs=site.data.menu}}
 
     {{>scripts}}
+    {{>registerTemplate}}
```
[the remark this answers is not in the document]
Comment (sarvaje) [reply src/hexo/themes/sonarwhal/layout/index.hbs]: Call me picky, but I think this name should be something more specific, like `registerWorker` or `registerWorkerTemplate`

[the remark this answers is not in the document]
Comment (molant) [reply src/hexo/themes/sonarwhal/layout/index.hbs]: Ideally https://github.com/sonarwhal/sonarwhal.com/pull/281#discussion_r147825455 should be part of `{{>scripts}}` and solve this issue as well :)

**Comment by molant** [comment]
@qzhou1607 is there anything remaining for this? First step was to get a service worker and the next one to add push notifications.

**Comment by qzhou1607** [comment]
@molant The service worker part is finished, so this PR is ready if we don't need push notifications for now.

**Comment by molant** [comment]
@sarvaje or @alrra can you take a look at this PR and merge if everything is 👌?

**Pull request merged**